Packs/Legacy/Integrations/Tanium/Tanium.py

```diff
@@ -41,13 +38,109 @@
+        current_cell: the value of the multi-lined single cell in the table
+
+    Returns:
+        Tuple(str, list). The full value of the single multi-lined cell and the rest of the values in it's line if exist
+    """
+    current_line = current_line + 1
+    line_below = lines[current_line].split(',')
+
+    # removing enter empty lines
+    if line_below[0] != '':
+        current_cell = current_cell + "\n" + line_below[0]
+
+    if line_below[0].endswith("\""):
+        # case1: no additional info on that line
+        if len(line_below) == 1:
+            return remove_apostrophes(current_cell), [], current_line
+
+        # case2: additional info on that line
+        else:
+            return remove_apostrophes(current_cell), line_below[1:], current_line
+
+    return get_full_cell(lines, current_line, current_cell)
+
+
+def gather_line(lines, current_line):
+    """gather the values of a single line in the answered csv
+
+    Args:
+        lines (list): a list of all lines in the csv
+        current_line (int): the current line being gathered
+
+    Returns:
+        Tuple(list, int). A list containing all the values of a single line, the number of the last line checked
+    """
+    line_values = lines[current_line].split(',')
+
+    # edge case - empty starting line
+    if len(line_values) == 0 or (len(line_values) == 1 and len(line_values[0]) == 0):
+        current_line += 1
+        return [], current_line
+
+    # multi lined cells start with " and end with "
+    while line_values[-1].startswith("\"") and not line_values[-1].endswith("\""):
+        current_cell = line_values[-1]
+        current_cell, rest_of_line, current_line = get_full_cell(lines, current_line, current_cell)
+        line_values[-1] = current_cell
+        line_values.extend(rest_of_line)
+
+    return line_values, current_line
+
+
+def csvstr_to_list(text_content):
+    """Gets a csv string representing a table of an answer given from Tanium and formats it as a list of dictionaries
+
+    Args:
+        text_content (str): a csv string containing the tabled answer for the question asked in Tanium
+
+    Returns:
+        list. The formatted answer as a list where each element is a dict representing a line of the tabled answer
+    """
+    lines = text_content.splitlines()
     if len(lines) < 2:
         return []
-    else:
-        headers = lines[0].split(',')
-        return [dict(zip(headers, line.split(','))) for line in lines[1:]]
+
+    headers = lines[0].split(',')
+    total_column_num = len(headers)
+    lines = lines[1:]
+    total_lines_num = len(lines)
+    result = []
+    current_line = 0
+    while current_line < total_lines_num:
+        line_dict = {}
+        line_values, current_line = gather_line(lines, current_line)
+
+        if len(line_values) == 0:
+            continue
+
+        for index in range(total_column_num):
+            line_dict[headers[index]] = line_values[index]
```

Comment:
why not `for header in headers`?

Reply:
not sure this looks better:
for header in headers:
     line_dict[header] = line_values[headers.index(header)]